### PR TITLE
fix for when warning is output

### PR DESCRIPTION
**Pull request by kab163**
### cmake -DENABLE_HOST_SHARED_MEMORY=On -DENABLE_MPI=On ../
-- The CXX compiler identification is GNU 8.3.1
-- The C compiler identification is GNU 8.3.1
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/c++
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/cc
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Using CMake version 3.14.5
-- BLT Version: 0.4.0
-- CMake Version: 3.14.5
-- CMake Executable: /usr/tce/packages/cmake/cmake-3.14.5/bin/cmake
-- Found Git: /usr/tcetmp/bin/git (found version "2.29.1")
-- Git Support is ON
-- Git Executable: /usr/tcetmp/bin/git
-- Git Version: 2.29.1
-- MPI Support is On
-- FindMPI Enabled  (ENABLE_FIND_MPI == ON)
-- Found MPI_C: /usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/lib/libmpiprofilesupport.so (found version "3.1")
-- Found MPI_CXX: /usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/lib/libmpiprofilesupport.so (found version "3.1")
-- Found MPI: TRUE (found version "3.1")
-- BLT MPI Compile Flags:
-- BLT MPI Include Paths:  /usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/include;/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/include
-- BLT MPI Libraries:      /usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/lib/libmpiprofilesupport.so;/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/lib/libmpi_ibm.so
-- BLT MPI Link Flags:     SHELL:-L/usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-2020.08.19/lib -pthread
-- MPI Executable:       /usr/tce/packages/spectrum-mpi/ibm/spectrum-mpi-rolling-release/bin/mpiexec
-- MPI Num Proc Flag:    -n
-- MPI Command Append:
-- OpenMP Support is Off
-- CUDA Support is Off
-- HIP Support is Off
-- HCC Support is OFF
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components:  doxygen dot
-- Sphinx support is ON
-- Found Sphinx: /usr/bin/sphinx-build
-- Valgrind support is ON
-- Found Valgrind: /usr/tcetmp/bin/valgrind
-- AStyle support is ON
-- Failed to locate AStyle executable (missing: ASTYLE_EXECUTABLE)
-- ClangFormat support is ON
-- Failed to locate ClangFormat executable (missing: CLANGFORMAT_EXECUTABLE)
-- Uncrustify support is ON
-- Failed to locate Uncrustify executable (missing: UNCRUSTIFY_EXECUTABLE)
-- Yapf support is ON
-- Failed to locate Yapf executable (missing: YAPF_EXECUTABLE)
-- CMakeFormat support is ON
-- Failed to locate CMakeFormat executable (missing: CMAKEFORMAT_EXECUTABLE)
-- Cppcheck support is ON
-- Failed to locate Cppcheck executable (missing: CPPCHECK_EXECUTABLE)
-- ClangQuery support is Off
-- ClangTidy support is ON
-- Failed to locate ClangTidy executable (missing: CLANGTIDY_EXECUTABLE)
-- C Compiler family is GNU
-- Adding optional BLT definitions and compiler flags
-- Setting CMAKE_CXX_EXTENSIONS to Off
-- Standard C++11 selected
-- Enabling all compiler warnings on all targets.
-- Fortran support disabled.
-- CMAKE_C_FLAGS flags are:    -Wall -Wextra
-- CMAKE_CXX_FLAGS flags are:       -Wall -Wextra
-- CMAKE_EXE_LINKER_FLAGS flags are:
-- Google Test Support is ON
-- Google Mock Support is On
-- Found PythonInterp: /usr/tcetmp/bin/python (found version "2.7.16")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Memcheck suppressions file: /g/g0/belcher6/Umpire/cmake/valgrind.supp
-- Setting C standard to 99
-- Checking for std::filesystem
-- Performing Test UMPIRE_ENABLE_FILESYSTEM
-- Performing Test UMPIRE_ENABLE_FILESYSTEM - Failed
-- std::filesystem NOT found, using POSIX
-- Host Shared Memory Enabled
-- Configuring done
-- Generating done
-- Build files have been written to: /g/g0/belcher6/Umpire/build

### cmake -DENABLE_TOOLS=On ../
-- The CXX compiler identification is GNU 8.3.1
-- The C compiler identification is GNU 8.3.1
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/c++
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/cc
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.3.1/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Using CMake version 3.14.5
-- BLT Version: 0.4.0
-- CMake Version: 3.14.5
-- CMake Executable: /usr/tce/packages/cmake/cmake-3.14.5/bin/cmake
-- Found Git: /usr/tcetmp/bin/git (found version "2.29.1")
-- Git Support is ON
-- Git Executable: /usr/tcetmp/bin/git
-- Git Version: 2.29.1
-- MPI Support is OFF
-- OpenMP Support is Off
-- CUDA Support is Off
-- HIP Support is Off
-- HCC Support is OFF
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components:  doxygen dot
-- Sphinx support is ON
-- Found Sphinx: /usr/bin/sphinx-build
-- Valgrind support is ON
-- Found Valgrind: /usr/tcetmp/bin/valgrind
-- AStyle support is ON
-- Failed to locate AStyle executable (missing: ASTYLE_EXECUTABLE)
-- ClangFormat support is ON
-- Failed to locate ClangFormat executable (missing: CLANGFORMAT_EXECUTABLE)
-- Uncrustify support is ON
-- Failed to locate Uncrustify executable (missing: UNCRUSTIFY_EXECUTABLE)
-- Yapf support is ON
-- Failed to locate Yapf executable (missing: YAPF_EXECUTABLE)
-- CMakeFormat support is ON
-- Failed to locate CMakeFormat executable (missing: CMAKEFORMAT_EXECUTABLE)
-- Cppcheck support is ON
-- Failed to locate Cppcheck executable (missing: CPPCHECK_EXECUTABLE)
-- ClangQuery support is Off
-- ClangTidy support is ON
-- Failed to locate ClangTidy executable (missing: CLANGTIDY_EXECUTABLE)
-- C Compiler family is GNU
-- Adding optional BLT definitions and compiler flags
-- Setting CMAKE_CXX_EXTENSIONS to Off
-- Standard C++11 selected
-- Enabling all compiler warnings on all targets.
-- Fortran support disabled.
-- CMAKE_C_FLAGS flags are:    -Wall -Wextra
-- CMAKE_CXX_FLAGS flags are:       -Wall -Wextra
-- CMAKE_EXE_LINKER_FLAGS flags are:
-- Google Test Support is ON
-- Google Mock Support is On
-- Found PythonInterp: /usr/tcetmp/bin/python (found version "2.7.16")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Memcheck suppressions file: /g/g0/belcher6/Umpire/cmake/valgrind.supp
-- Setting C standard to 99
-- Checking for std::filesystem
-- Performing Test UMPIRE_ENABLE_FILESYSTEM
-- Performing Test UMPIRE_ENABLE_FILESYSTEM - Failed
-- std::filesystem NOT found, using POSIX
-- Host Shared Memory Disabled
-- Looking for output.replay reference output...
-- Using output.replay as good replay test data
-- Configuring done
-- Generating done
-- Build files have been written to: /g/g0/belcher6/Umpire/build

### [belcher6@corona212:build]$ 
cmake -DENABLE_HIP=On -DHIP_ROOT_DIR="/opt/rocm-4.1.0/hip" -DHIP_HIPCC_FLAGS="--amdgpu-target=gfx906" -DHIP_RUNTIME_INCLUDE_DIRS="/opt/rocm-4.1.0/hip/include;/opt/rocm-4.1.0/hip/../hsa/include" -DHIP_CLANG_PATH="/opt/rocm-4.1.0/hip/../llvm/bin" -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_COMPILER="/usr/tce/packages/gcc/gcc-8.1.0/bin/g++" -DCMAKE_C_COMPILER="/usr/tce/packages/gcc/gcc-8.1.0/bin/gcc" -DHIP_CLANG_FLAGS="--gcc-toolchain=/usr/tce/packages/gcc/gcc-8.1.0/bin/.." -DCMAKE_EXE_LINKER_FLAGS="-Wl,--disable-new-dtags -L/opt/rocm-4.1.0/hip/lib -L/opt/rocm-4.1.0/lib -L/opt/rocm-4.1.0/lib64 -Wl,-rpath,/opt/rocm-4.1.0/hip/lib:/opt/rocm-4.1.0/lib:/opt/rocm-4.1.0/lib64 -lamdhip64 -lhsakmt -lhsa-runtime64 -Wl,-rpath /usr/tce/packages/gcc/gcc-8.1.0/bin/../lib64"  ../
-- The CXX compiler identification is GNU 8.1.0
-- The C compiler identification is GNU 8.1.0
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.1.0/bin/g++
-- Check for working CXX compiler: /usr/tce/packages/gcc/gcc-8.1.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.1.0/bin/gcc
-- Check for working C compiler: /usr/tce/packages/gcc/gcc-8.1.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Using CMake version 3.14.5
-- BLT Version: 0.4.0
-- CMake Version: 3.14.5
-- CMake Executable: /usr/tce/packages/cmake/cmake-3.14.5/bin/cmake
-- Found Git: /usr/tce/bin/git (found version "2.29.1")
-- Git Support is ON
-- Git Executable: /usr/tce/bin/git
-- Git Version: 2.29.1
-- MPI Support is OFF
-- OpenMP Support is Off
-- CUDA Support is Off
-- HIP Support is On
-- Found HIP: /opt/rocm-4.1.0/hip (found version "4.1.21072-c3eb5ccc")
-- HIP version:      4.1.21072-c3eb5ccc
-- HIP platform:     amd
-- HCC Support is OFF
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components:  doxygen dot
-- Sphinx support is ON
-- Found Sphinx: /usr/bin/sphinx-build
-- Valgrind support is ON
-- Found Valgrind: /usr/tce/bin/valgrind
-- AStyle support is ON
-- Failed to locate AStyle executable (missing: ASTYLE_EXECUTABLE)
-- ClangFormat support is ON
-- Failed to locate ClangFormat executable (missing: CLANGFORMAT_EXECUTABLE)
-- Uncrustify support is ON
-- Failed to locate Uncrustify executable (missing: UNCRUSTIFY_EXECUTABLE)
-- Yapf support is ON
-- Failed to locate Yapf executable (missing: YAPF_EXECUTABLE)
-- CMakeFormat support is ON
-- Failed to locate CMakeFormat executable (missing: CMAKEFORMAT_EXECUTABLE)
-- Cppcheck support is ON
-- Failed to locate Cppcheck executable (missing: CPPCHECK_EXECUTABLE)
-- ClangQuery support is Off
-- ClangTidy support is ON
-- Failed to locate ClangTidy executable (missing: CLANGTIDY_EXECUTABLE)
-- C Compiler family is GNU
-- Adding optional BLT definitions and compiler flags
-- Setting CMAKE_CXX_EXTENSIONS to Off
-- Standard C++11 selected
-- Enabling all compiler warnings on all targets.
-- Fortran support disabled.
-- CMAKE_C_FLAGS flags are:    -Wall -Wextra
-- CMAKE_CXX_FLAGS flags are:       -Wall -Wextra
-- CMAKE_EXE_LINKER_FLAGS flags are:  -Wl,--disable-new-dtags -L/opt/rocm-4.1.0/hip/lib -L/opt/rocm-4.1.0/lib -L/opt/rocm-4.1.0/lib64 -Wl,-rpath,/opt/rocm-4.1.0/hip/lib:/opt/rocm-4.1.0/lib:/opt/rocm-4.1.0/lib64 -lamdhip64 -lhsakmt -lhsa-runtime64 -Wl,-rpath /usr/tce/packages/gcc/gcc-8.1.0/bin/../lib64
-- Google Test Support is ON
-- Google Mock Support is On
-- Found PythonInterp: /usr/tce/bin/python (found version "2.7.16")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Memcheck suppressions file: /g/g0/belcher6/Umpire/cmake/valgrind.supp
-- Setting C standard to 99
-- Checking for std::filesystem
-- Performing Test UMPIRE_ENABLE_FILESYSTEM
-- Performing Test UMPIRE_ENABLE_FILESYSTEM - Failed
-- std::filesystem NOT found, using POSIX
-- Host Shared Memory Disabled
-- **Disabling replay tests for HIP build.**
-- Configuring done
-- Generating done
-- Build files have been written to: /g/g0/belcher6/Umpire/build

### CODE REVIEW

File: tests/integration/CMakeLists.txt [hunk 1]
```diff
@@ -210,7 +210,9 @@ endif ()
 if (ENABLE_TOOLS AND NOT ENABLE_HIP)
   add_subdirectory(replay)
 else ()
-  message(STATUS "Disabling replay tests for HIP build.")
+  if (ENABLE_HIP)
```
Comment (davidbeckingsale) [inline tests/integration/CMakeLists.txt:213]:
```suggestion
  if (ENABLE_HIP AND ENABLE_TOOLS)
```

**Comment by kab163** [comment]
Description is showing that the warning only is output when intended

File: tests/integration/CMakeLists.txt (hunk 1 above)
Comment (davidbeckingsale) [reply tests/integration/CMakeLists.txt:213]: So the warning only prints if tools were enabled.

Comment (kab163) [reply tests/integration/CMakeLists.txt:213]: Oh jeez... isn't there a better way to write that if statement?
```
if (enable_tools)
  if (not enable_hip)
    add the sub dir
  else
    warning
  endif
endif
```

Comment (davidbeckingsale) [reply tests/integration/CMakeLists.txt:213]: Yeah, my brain is not working 😆

**Review by davidbeckingsale — APPROVED** [review]

**Pull request merged**